fleetctl/fleetctl.go

```diff
@@ -1060,6 +1061,87 @@ func assertUnitState(name string, js job.JobState, out io.Writer) (ret bool) {
 	return
 }
 
+// tryWaitForSystemdActiveState tries to wait for systemd units to reach an
+// active state, making use of the systemd's DBUS interface. First it takes one
+// or more units as input, and ensures that every unit in the []units must be
+// in the active state. If yes, return 0. Otherwise return 1.
+//
+// NOTE: To avoid unnecessary conflicts with unit tests in start_test.go, we
+// should return 0 for the case of sd_dbus.New() returning error. In that case,
+// it just means that the unit test environment based on Travis is not capable
+// of setting up a DBUS connection of systemd, which would actually work fine
+// for production systems or functional tests.
+func tryWaitForSystemdActiveState(units []string) (ret int) {
```

Comment:
imho the ret value is pretty unidiomatic; rather, return an error, and have the callers decide to log the error or not